imap_processing/ialirt/l0/parse_mag.py

```diff
@@ -0,0 +1,246 @@
+
+    Parameters
+    ----------
+    val : int
+        24-bit integer value.
+
+    Returns
+    -------
+    list[int]
+        List of three extracted bytes.
+    """
+    return [
+        (val >> 16) & 0xFF,  # Most significant byte (Byte2)
+        (val >> 8) & 0xFF,  # Middle byte (Byte1)
+        (val >> 0) & 0xFF,  # Least significant byte (Byte0)
+    ]
+
+
+def extract_magnetic_vectors(science_values: xr.DataArray) -> dict:
+    """
+    Extract the magnetic vectors.
+
+    Parameters
+    ----------
+    science_values : xr.DataArray
+        Science data.
+
+    Returns
+    -------
+    vectors : dict
+        Magnetic vectors.
+    """
+    # Convert each 24-bit value to its three constituent bytes
+    science0 = get_bytes(int(science_values[0]))
+    science1 = get_bytes(int(science_values[1]))
+    science2 = get_bytes(int(science_values[2]))
+    science3 = get_bytes(int(science_values[3]))
+
+    # Primary sensor:
+    pri_x = (science0[0] << 8) | science0[1]
+    pri_y = (science0[2] << 8) | science1[0]
+    pri_z = (science1[1] << 8) | science1[2]
+
+    # Secondary sensor:
+    sec_x = (science2[0] << 8) | science2[1]
+    sec_y = (science2[2] << 8) | science3[0]
+    sec_z = (science3[1] << 8) | science3[2]
+
+    vectors = {
+        "pri_x": pri_x,
+        "pri_y": pri_y,
+        "pri_z": pri_z,
+        "sec_x": sec_x,
+        "sec_y": sec_y,
+        "sec_z": sec_z,
+    }
+
+    return vectors
+
+
+def get_time(grouped_data: xr.Dataset, group: int, pkt_counter: xr.DataArray) -> dict:
+    """
+    Get the time for the grouped data.
+
+    Parameters
+    ----------
+    grouped_data : xr.Dataset
+        Grouped data.
+    group : int
+        Group number.
+    pkt_counter : xr.DataArray
+        Packet counter.
+
+    Returns
+    -------
+    time_data : dict
+        Coarse and fine time for Primary and Secondary Sensors.
+    """
+    pri_coarsetm = grouped_data["mag_acq_tm_coarse"][
+        (grouped_data["group"] == group).values
+    ][pkt_counter == 0]
+
+    pri_fintm = grouped_data["mag_acq_tm_fine"][
+        (grouped_data["group"] == group).values
+    ][pkt_counter == 0]
+
+    sec_coarsetm = grouped_data["mag_acq_tm_coarse"][
+        (grouped_data["group"] == group).values
+    ][pkt_counter == 2]
+
+    sec_fintm = grouped_data["mag_acq_tm_fine"][
+        (grouped_data["group"] == group).values
+    ][pkt_counter == 2]
+
+    time_data = {
+        "pri_coarsetm": int(pri_coarsetm),
+        "pri_fintm": int(pri_fintm),
+        "sec_coarsetm": int(sec_coarsetm),
+        "sec_fintm": int(sec_fintm),
+    }
+
+    return time_data
+
+
+def parse_packet(accumulated_data: xr.Dataset) -> list[dict]:
+    """
+    Parse the MAG packets.
+
+    Parameters
+    ----------
+    accumulated_data : xr.Dataset
+        Packets dataset accumulated over 1 min.
+
+    Returns
+    -------
+    mag_data : list[dict]
+        Dictionaries of the parsed data product.
+    """
+    logger.info("Parsing MAG.")
```

Comment:
Might be useful to put some additional info here like: `for time period: accumulated_data["mag_acq_tm_coarse"]`